### ISSUE TITLE
PERF102 - False Positive with Poorly Named Variables

### ISSUE THREAD

**Issue by Skylion007** [astral-sh/ruff#5692]
```
-                    for (i, j), _ in smat.items():
+                    for (i, j) in smat.keys():
                         diag_entries[(i + rmax, j + cmax)] = _
```
This code snippit is from sympy/matrices/common.py, this uses the latest stable ruff version and just has PERF102 manually selected using the command `ruff --select PERF --fix .`

I know that the `_` prefix is often used to look for variables that should be unused in a loop, but ruff should check if the variable is actually unused before autofixing it to prevent bugprone autofixes. It should at least handle this trivial case.

**Comment by charliermarsh** [comment]
Yeah we can support this.

**Comment by zanieb** [comment]
Wow that's madness :) We should add a rule that checks for usage of `_` outside of unpacking!

**Comment by dhruvmanila** [comment]
Yeah, I've seen this kind of usage in the wild especially in comprehensions for one off variables.

**Comment by charliermarsh** [comment]
`crates/ruff/src/rules/flake8_bugbear/rules/unused_loop_control_variable.rs` is a similar rule (it detects unused loop control variables more generally) that we can use as a reference.
